src/zoo.js

```diff
@@ -8,59 +8,104 @@ eslint no-unused-vars: [
   }
 ]
 */
-
 const data = require('./data');
 
-function animalsByIds(ids) {
-  // seu código aqui
+function animalsByIds(...ids) {
+  if (!ids) return [];
+  return ids.map(elemento => data.animals.find(arr => (arr.id === elemento)));
 }
 
-function animalsOlderThan(animal, age) {
-  // seu código aqui
+function animalsOlderThan(animal, idade) {
+  const filterFunction = data.animals.find(arr => arr.name === animal);
```

Comment:
o data.animals é uma array! Mas os elementos dessa array são objetos. 
Toda array é objeto, mas nem todo objeto é array! hahaha
Podemos imaginar que array é um objeto especial.

portanto quando usou o data.animals.find você vai iterar na array data.animals que tem como elementos objetos. Tanto que depois você usa arr.name. Se fosse uma array não funcionária o .name

```suggestion
function animalsOlderThan(animal, idade) {
  const filterFunction = data.animals.find(animalObj => animalObj.name === animal);
```